LayoutTests/inspector/model/source-map-spec.html

```diff
@@ -0,0 +1,112 @@
+<!DOCTYPE html>
+<html>
+<head>
+<script src="../../http/tests/inspector/resources/inspector-test.js"></script>
+<script>
+const baseSourceMapTestDir = "../../imported/tc39-tg4/source-map-tests"
+function triggerScriptResource(scriptName) {
+    let script = document.createElement("script");
+    script.src = `${baseSourceMapTestDir}/resources/${scriptName}`;
+    document.body.appendChild(script);
+}
+
+async function setup()
+{
+    const response = await fetch(`${baseSourceMapTestDir}/source-map-spec-tests.json`);
+    const json = await response.json();
+    TestPage.dispatchEventToFrontend("TestDescriptionsFetched", {json});
+}
+
+function test()
+{
+    function checkMapping(resource, testCase, action)
+    {
+        const location = resource.createSourceCodeLocation(action.generatedLine, action.generatedColumn);
+        InspectorTest.expectThat(location.hasMappedLocation(), `Test location (${action.generatedLine}, ${action.generatedColumn}) should be mapped`);
+        InspectorTest.expectEqual(location.displayLineNumber, action.originalLine, `Original line: ${location.displayLineNumber}, expected: ${action.originalLine}`);
+        InspectorTest.expectEqual(location.displayColumnNumber, action.originalColumn, `Original column: ${location.displayColumnNumber}, expected: ${action.originalColumn}`);
+        InspectorTest.expectEqual(location.displaySourceCode.displayName, action.originalSource, `Original source: ${location.displaySourceCode.displayName}, expected: ${action.originalSource}`);
+    }
+
+    function checkIgnoreList(resource, testCase, action)
+    {
+        const sourceMap = resource.sourceMaps[0];
+        InspectorTest.assert(sourceMap);
+        const result = action.present.every((sourceName) => {
+            return sourceMap.resources.some((resource) => {
+                return resource.displayName.match(sourceName);
+            });
+        });
+        InspectorTest.expectThat(result, "Ignored sources should be in ignoreList");
+    }
+
+    function withTimeout(promise, seconds)
+    {
+        return Promise.race([promise, new Promise((resolve, reject) => {
+            setTimeout(resolve, seconds);
+        })]);
+    }
+
+    let suite = InspectorTest.createAsyncSuite("SourceMapSpec");
+
+    suite.addTestCase({
+        name: "SourceMapSpec",
```

Comment:
on average how long does this test take to run?

all Web Inspector test cases have a default timeout of 10s, so will this run up against that or is it quicker?

have you tried running it in debug and/or using WK1 (i.e. legacy WebKit)?